lib/Frontend/CachingUtils.cpp

```diff
@@ -281,7 +284,17 @@ bool replayCachedCompilerOutputs(
                     toString(File.takeError()));
       continue;
     }
-    *File << Output.Proxy.getData();
+
+    if (UseCASBackend && Output.Kind == file_types::ID::TY_Object) {
+      auto Schema = std::make_unique<llvm::mccasformats::v1::MCSchema>(CAS);
+      SmallString<50> ContentsStorage;
+      llvm::raw_svector_ostream OS(ContentsStorage);
+      if (auto E = Schema->serializeObjectFile(Output.Proxy, OS))
+        llvm::report_fatal_error(std::move(E));
```

Comment:
Do not fatal_error in a function that return Error.

Reply:
I am confused, the function returns a bool, should I be using `Diag.diagnose` instead?